staff/sergio-costa/arshop/arshop-api/src/logic/index.js

```diff
@@ -0,0 +1,360 @@
+'use strict'
+
+const { models: { User, Product } } = require('arshop-data')
+const bcrypt = require('bcrypt')
+
+/**
+ * Abstraction of business logic.
+ */
+const logic = {
+    /**
+    * Registers a user.
+    * 
+    * @param {string} name 
+    * @param {string} surname 
+    * @param {string} email 
+    * @param {string} password 
+    * @param {string} passwordConfirmation 
+    */
+    registerUser(name, surname, email, password, passwordConfirmation) {
+        if (typeof name !== 'string') throw TypeError(name + ' is not a string')
```

Comment:
SEE `staff/manuel-barzi/music-validations'. may you use a module like that to apply validations in a generic way as in `staff/manuel-barzi/staff/music-api/src/logic/index.js`. check it out!